documentation/modules/configuring/proc-loading-config-with-provider.adoc

```diff
@@ -0,0 +1,152 @@
+// Module included in the following assemblies:
+//
+// assembly-config-kafka-connect.adoc
+
+[id='proc-loading-config-with-provider-{context}']
+= Loading configuration values from external sources
+
+[role="_abstract"]
+Use the _Kubernetes Configuration Provider_ plugin to load configuration data from external sources.
+Load data from Kubernetes Secrets or ConfigMaps.
+
+Suppose you have a Secret that's managed outside the Kafka namespace, or outside the Kafka cluster.
+The provider allows you to reference the values of the Secret in your configuration without extracting the files.
+You just need to tell the provider what Secret to use and provide access rights.
+The provider loads the data without needing to restart the Kafka component.
+
+The provider operates independently of Strimzi.
+You can use it to load configuration data for all Kafka components, including producers and consumers.
+Use it, for example, to provide the credentials for Kafka Connect connector configuration.
+
+In this procedure, an external ConfigMap provides configuration properties for a connector.
+
+NOTE: As an alternative to using the Kubernetes Configuration Provider, you can mount ConfigMaps or Secrets into a Kafka Connect pod as environment variables or volumes.
+You do this by adding configuration using the  xref:type-ExternalConfiguration-reference[`externalConfiguration` property] in `KafkaConnect.spec`.
```

Comment:
@scholzj I want to say why we would the Kubernetes provider over the mounting approach. Mounting is more disruptive? Better in certain situations?

Reply:
The main advantages of the Kubernetes Configuration provider are:
* Unlike the `externalConfiguration`, it does not need to restart the Kafka Connect to use new Secret or ConfigMap. This is important especially when sharing the Connect instance with multiple connectors, because there, mounting a secret in `externalConfiguration` for one connector would also disrupt all other connectors.
* It can use Secrets / ConfigMaps from different namespaces as well (assuming it gets the RBAC rights for it)

The disadvantages are:
* Need to setup the additional RBAC rights
* It does not allow to mount files, only to set the values directly into the configuration options. So it cannot be used for example to configure options which require a location of a truststore or keystore.

Reply:
Thanks. I've rewritten.